src/core/tests/Test_StatusHandlerTruncation.py

```diff
@@ -0,0 +1,960 @@
+
+        # Set up complete status file before exceptions
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.complete_status_file_path, 'r') as file_handle:
+            complete_substatus_file_data = json.load(file_handle)
+
+        # Assert complete status file size > 128kb and no exceptions
+        self.assertTrue(len(json.dumps(complete_substatus_file_data[0]["status"]["substatus"][0])) > Constants.StatusTruncationConfig.AGENT_FACING_STATUS_FILE_SIZE_LIMIT_IN_BYTES)
+        message = self.__get_message_json_from_substatus(complete_substatus_file_data)
+        self.assertEqual(len(message["errors"]["details"]), 0)
+
+        # Set up complete status file after exceptions
+        self.__add_multiple_exception_errors()
+        self.runtime.status_handler.set_installation_substatus_json(status=Constants.STATUS_ERROR)
+
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.complete_status_file_path, 'r') as file_handle:
+            complete_substatus_file_data = json.load(file_handle)
+        self.__assert_status_file_data_multi_exception_errors(complete_substatus_file_data, Constants.PATCH_INSTALLATION_SUMMARY, Constants.STATUS_ERROR, patch_count,
+            count_errors_detail=5)
+
+        # Assert installation truncated status file with multi exceptions
+        self.__assert_truncated_status_multi_exceptions(complete_substatus_file_data, Constants.PATCH_INSTALLATION_SUMMARY, Constants.STATUS_ERROR, patch_count, errors_count=5)
+
+    def test_both_assessment_and_installation_over_size_limit_truncated(self):
+        """ Perform truncation on very large assessment patches for time performance.
+        Output:
+        Input: xxx assessment patches
+        Output:
+        operation: Installation,
+        assessment substatus name: PatchAssessmentSummary,
+        complete status file size > 128kb,
+        truncated status file size < 126kb < 128kb,
+        truncated status file size < completed status file size,
+        assessment substatus status: warning,
+        assessment substatus patches < patch_count,
+        assessment errors code: 2 (warning),
+        assessment errors details count: 0,
+        assessment message json fields == truncated assessment message json fields.
+
+        Perform truncation on very large installation patches for time performance.
+        Input: xxx installation patches
+        Output:
+        operation: Installation,
+        installation substatus name: PatchInstallationSummary,
+        complete status file size > 128kb,
+        truncated status file size < 126kb < 128kb,
+        truncated status file size < completed status file size,
+        installation substatus status: warning,
+        installation errors code: 2 (warning),
+        installation errors details count: 0,
+        installation message json fields == truncated installation message json fields. """
```

Comment:
Combine input and output description.

Reply:
I tried to shorten the description/input/ouput as much as possible. can you give me an example of it if it doesn't meet your expectations?

done